src/hyperion/external_interaction/callbacks/xray_centre/ispyb_callback.py

```diff
@@ -97,35 +105,38 @@ def activity_gated_start(self, doc: RunStart):
                 self.params.hyperion_params.detector_params,
                 self.params.hyperion_params.ispyb_params,
             )
-            grid_scan_info = GridScanInfo(
-                self.params.hyperion_params.ispyb_params.upper_left,
-                self.params.experiment_params.y_steps,
-                self.params.experiment_params.y_step_size,
-            )
 
-            def constructor():
-                return construct_comment_for_gridscan(
-                    self.params,
-                    self.params.hyperion_params.ispyb_params,
-                    grid_scan_info,
-                )
-
-            scan_data_info = ScanDataInfo(
-                data_collection_info=populate_remaining_data_collection_info(
-                    constructor,
-                    None,
-                    populate_xy_data_collection_info(
-                        grid_scan_info,
-                        self.params,
-                        self.params.hyperion_params.ispyb_params,
+            scan_data_infos = [
+                ScanDataInfo(
+                    data_collection_info=populate_remaining_data_collection_info(
+                        None,
+                        None,
```

Comment:
Should: Shouldn't the `dcgid` here be the group made above?

Reply:
we haven't made it yet, we've only populated the `DataCollectionGroupInfo` but it hasn't been posted to the database so there's no id, leaving it as none will cause it to be populate during the call to `begin_deposition`